sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java

```diff
@@ -175,10 +176,15 @@ public Mono<RxDocumentServiceRequest> populateFeedRangeFilteringHeaders(
                         }
 
                         final List<PartitionKeyRange> pkRanges = pkRangeHolder.v;
-                        if (pkRanges == null || pkRanges.size() == 0) {
+                        if (pkRanges == null) {
                             return Mono.error(new NotFoundException(
-                                String.format("Stale cache for collection rid '%s'.",
-                                    containerRid)));
+                                String.format("Stale cache for collection rid '%s': pkRanges are null",
```

Comment:
In both cases I think we should also log this.range?

Reply:
added